src/core/geometry/qgsregularpolygon.h

```diff
@@ -0,0 +1,213 @@
+/***************************************************************************
+                         qgsregularpolygon.h
+                         --------------
+    begin                : May 2017
+    copyright            : (C) 2017 by Loîc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSREGULARPOLYGON_H
+#define QGSREGULARPOLYGON_H
+
+#include <QString>
+
+#include "qgis_core.h"
+#include "qgspointv2.h"
+#include "qgspolygon.h"
+#include "qgslinestring.h"
+#include "qgscircle.h"
+#include "qgstriangle.h"
+
+enum ConstructionOption
+{
+  inscribedCircle,
+  circumscribedCircle
+};
+
+/** \ingroup core
+ * \class QgsRegularPolygon
+ * \brief Regular Polygon geometry type.
+ *
+ * A regular polygon is defined by a center point with a number of sides/vertices, a radius and the first vertice.
+ *
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsRegularPolygon
+{
+  public:
+    QgsRegularPolygon();
+
+    /** Constructs a regular polygon by center and parameters for the first vertice. An empty regular polygon is returned if numSides < 3 or circle option isn't valid.
```

Comment:
if \a numSides < 3

the \a annotation will cause numSides to be formatted as an argument in the doxygen pages